sink/sink

```diff
@@ -434,6 +484,42 @@ def main():
 
     # Create the directory and chdir
     logs = os.path.expanduser(config.get("Sink", "Logs"))
+
+    # See if we have to prune logs
+    prune_interval = config.get("Sink", "PruneInterval")
+    if prune_interval >= 0:
+        status_file = os.path.expanduser(config.get("Sink", "LogsPruneStatus"))
+        prune_now = True
+        # use the last modified time of the logfile itself, not just the directory
```

Comment:
We should break this stuff out into its own function, because this function is long enough.